pkg/plugins/builtin/http/http.go

```diff
@@ -132,6 +134,19 @@ func exec(stepName string, config interface{}, ctx interface{}) (interface{}, in
 		return nil, nil, fmt.Errorf("HTTP request failed: %s", err.Error())
 	}
 
+	// remove response magic prefix
+	if cfg.MagicPrefix != "" {
+		respBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, nil, fmt.Errorf("HTTP cannot read response: %s", err.Error())
+		}
+		resp.Body.Close()
+		if bytes.HasPrefix(respBody, []byte(cfg.MagicPrefix)) {
+			respBody = bytes.Replace(respBody, []byte(cfg.MagicPrefix), nil, 1)
```

Comment:
I'd suggest
```
respBody = respBody[len(cfg.MagicPrefix):]
```
to avoid recopying the whole underlying array.

Maybe store cfg.MagicPrefix as a byte slice like cfg.MagicPrefixBytes ? To avoid casting string to []byte every time, which also does a copy.

Reply:
done ( but just create a local variable to store the byte slice )